[enocean] D2-50 EEP remove extra channels and fix warnings

lovery opened the pull request:
# Description
Improvement: Fix 7 warnings on building a jar:
* `Null comparison always yields false...` - `EEPFactory.java:[54,17]`, `EEPFactory.java:[67,17]`, `EnOceanBaseSensorHandler.java:[158,13]`
* `Dead code` (caused by the above warinings) - `EEPFactory.java:[54,29]`, `EEPFactory.java:[67,29]`, `EnOceanBaseSensorHandler.java:[158,41]`
* `Potential null pointer access` - `EnOceanTransceiver.java:[451,13]`
* `Redundant null check: The variable [name] cannot be null at this location` - `EnOceanDeviceDiscoveryService.java:[158,17]`, `A5_3F_7F_EltakoFSB.java:[70,24]`, `EEPType.java:[750,17]`
* `The nullness annotation is redundant with a default that applies to this location` - `EnOceanDeviceDiscoveryService.java:[88,16]`


Improvement: remove channels which are not supported by device using EEP D2-50-00, D2-50-01, D2-50-10, D2-50-11.
Before this PR choosing either of the above EEP doesn't change anything in the channels list, it looked like all the devices support all channels, but according to the specification they don't. For reference check EnOcean specification on [page 236 ](https://www.enocean-alliance.org/wp-content/uploads/2017/05/EnOcean_Equipment_Profiles_EEP_v2.6.7_public.pdf)

![Screenshot from 2024-10-09 10-51-00](https://github.com/user-attachments/assets/a2de6bcd-d175-4437-98a9-bc49a1df0d41)


Test [jar file](https://drive.google.com/file/d/19IFIxJyUPJ-4LgWY2bOhRaYxtJPN_5EM/view?usp=sharing)

lsiepel reviewed (changes requested):
Thanks, otherwise LGTM

lovery commented on bundles/org.openhab.binding.enocean/src/main/java/org/openhab/binding/enocean/internal/eep/EEPFactory.java:
```diff
@@ -51,7 +52,7 @@ public class EEPFactory {
     public static EEP createEEP(EEPType eepType) {
         try {
             Class<? extends EEP> cl = eepType.getEEPClass();
-            if (cl == null) {
+            if (Objects.isNull(cl)) {
```
I wanted to lower the amount of warnings on building and with the code before I was getting
```
[WARNING] /home/zhivka/workspace/openhab/openhab-addons/bundles/org.openhab.binding.enocean/src/main/java/org/openhab/binding/enocean/internal/eep/EEPFactory.java:[54,17] Null comparison always yields false: The variable cl cannot be null at this location
[WARNING] /home/zhivka/workspace/openhab/openhab-addons/bundles/org.openhab.binding.enocean/src/main/java/org/openhab/binding/enocean/internal/eep/EEPFactory.java:[54,29] Dead code
```
After my change those warnings didn't appear. Do you know better way to fix those warnings?

lsiepel replied:
According to the `getEEPClass()` annotation it will never return null, so the compiler knows this is dead code.
So either these checks can be removed OR the annotation should be changed. I'm 99% sure its the first.

lovery commented:
@lsiepel I followed your advice and removed the check for objects that can't be null and removed the use of `Objects.isNull`

I also added one more commit with fixes for `Redundant null check: The variable [name] cannot be null at this location`.
There is one more place that I don't know how to fix it: 
https://github.com/openhab/openhab-addons/blob/a6bb5373ea21fabbc55b5319c5c1e45547ead3d5/bundles/org.openhab.binding.enocean/src/main/java/org/openhab/binding/enocean/internal/handler/EnOceanBaseSensorHandler.java#L88-L92

The warning looks like this: `.../EnOceanBaseSensorHandler.java:[91,21] Redundant null check: The variable existingKey cannot be null at this location`
`receivingEEPTypes` is from type `Hashtable<RORG, EEPType>` and as far as I understand the function `putIfAbsent` will return null if the key is already present and it will return the value if not, but since it is assigned to a variable which cant be null what will happen? Do you have recommendation how to fix this or if you can point me in a direction where to search for a solutions?

lsiepel commented:
> putIfAbsent

These annotations are not perfect :-/ `putIfAbsent` is a known issue to the current used lastnpe library. These annotatiosn are nothing more then hints, so the actual code is more important. In this case the warning is a false positive and i find it acceptable to merge.

lsiepel reviewed (approved):
Thanks, LGTM
